From 3c9538debdeb5bd7cbcc09d2c0c2b0f40edd95f7 Mon Sep 17 00:00:00 2001
From: grotaclas <grotaclas@gmail.com>
Date: Fri, 26 Dec 2025 00:09:41 -0000
Subject: [PATCH] Restrict trigger of flavor_brapru.1 to prevent unintenional
 situations

The event "The incorporation of Prussia into Brandenburg" (flavor_brapru.1)
turns Prussia into a junior partner of Brandenburg. This patch adds
additional requirements:

* both countries are christian
* both countries are at peace(to avoid weir interactions if they are
  both in the same war and to avoid taking Prussia out of a war in which
the player intended to take something from them)
* Brandenburg must be valid_for_personal_unions_trigger(especially it
  must not be a subject, because otherwise Prussia would become the
junior partner of Brandenburg's overlord instead)
* Prussia must either be is_free_or_tributary_trigger or a vassal(but
  not a vassal of Brandenburg). This allows the historical situation
that Prussia is a vassal of Poland/PLC, and allows taking them away from
another country which takes Poland's place, but it prevents possibly
more problematic interactions if they are another subject type. E.g. the
event mentions that Prussia has been ruled by Hohenzollern's for
generations so it would be weird taking it if it is a junior partner of
another country.

A possible improvement which I didn't implement would be to require that
Prussia has at least 50% liberty desire if they are a subject.

Bugreport: https://forum.paradoxplaza.com/forum/threads/getting-the-brandenburg-prussia-pu-event-as-sweden-while-having-both-as-vassals.1697129/
---
 events/flavorBRAPRU.txt | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/events/flavorBRAPRU.txt b/events/flavorBRAPRU.txt
index 43d99a8b..61761301 100755
--- a/events/flavorBRAPRU.txt
+++ b/events/flavorBRAPRU.txt
@@ -20,14 +20,23 @@ country_event = {
 	
 	trigger = {
 		tag = BRA
-		government = monarchy
+		valid_for_personal_unions_trigger = yes
+		is_at_war = no
 		PRU = {
 			government = monarchy
 			ai = yes
 			was_player = no
 			NOT = { num_of_subjects = 1 }
 			has_opinion = { who = BRA value = 100 }
-			NOT = { vassal_of = BRA }
+			is_at_war = no
+			religion_group = christian
+			OR = {
+				is_free_or_tributary_trigger = yes
+				AND = {
+					is_vassal = yes
+					NOT = { vassal_of = BRA }
+				}
+			}
 		}
 	}
 	
-- 
2.51.0

